crates/op-rbuilder/src/builders/flashblocks/ctx.rs

```diff
@@ -0,0 +1,75 @@
+use crate::{
+    builders::{BuilderConfig, OpPayloadBuilderCtx, flashblocks::FlashblocksConfig},
+    gas_limiter::AddressGasLimiter,
+    metrics::OpRBuilderMetrics,
+    traits::ClientBounds,
+};
+use op_revm::OpSpecId;
+use reth_basic_payload_builder::PayloadConfig;
+use reth_evm::EvmEnv;
+use reth_optimism_chainspec::OpChainSpec;
+use reth_optimism_evm::{OpEvmConfig, OpNextBlockEnvAttributes};
+use reth_optimism_payload_builder::{OpPayloadBuilderAttributes, config::OpDAConfig};
+use reth_optimism_primitives::OpTransactionSigned;
+use std::sync::Arc;
+use tokio_util::sync::CancellationToken;
+
+#[derive(Debug, Clone)]
+pub(super) struct OpPayloadSyncerCtx {
+    /// The type that knows how to perform system calls and configure the evm.
+    evm_config: OpEvmConfig,
+    /// The DA config for the payload builder
+    da_config: OpDAConfig,
+    /// The chainspec
+    chain_spec: Arc<OpChainSpec>,
+    /// Max gas that can be used by a transaction.
+    max_gas_per_txn: Option<u64>,
+}
+
+impl OpPayloadSyncerCtx {
+    pub(super) fn new<Client>(
+        client: &Client,
+        builder_config: BuilderConfig<FlashblocksConfig>,
+        evm_config: OpEvmConfig,
+    ) -> eyre::Result<Self>
+    where
+        Client: ClientBounds,
+    {
+        let chain_spec = client.chain_spec();
+        Ok(Self {
+            evm_config,
+            da_config: builder_config.da_config.clone(),
+            chain_spec,
+            max_gas_per_txn: builder_config.max_gas_per_txn,
+        })
+    }
+
+    pub(super) fn evm_config(&self) -> &OpEvmConfig {
+        &self.evm_config
+    }
+
+    pub(super) fn into_op_payload_builder_ctx(
```

Comment:
why not <Into> for conversions between the types?

Reply:
i'm probably going to remove this function after cleanup